Reservation service validation

bryan-wood48 opened the pull request:
pull request reopened after minor error correction

Commit:
Update ReservationServiceImpl.java

completed reservation date validation

Commit:
Update ReservationServiceImpl.java

Commit:
Update ReservationServiceImpl.java

Commit:
Update ReservationServiceImpl.java

minor edit to correct comparators

zsherry0803 reviewed (commented):
from line 15 to line 83. Why you add extra space on every coded line? And the space does not apply to our coding-style after you changed. Please review it or let me know if I missed anything.

Commit:
Update ReservationServiceImpl.java

corrected the comparator methods to utilize class comparators for simple date/time validation
